src/Umbraco.PublishedCache.NuCache/PublishedSnapshotStatus.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Umbraco.Infrastructure.PublishedCache.Persistence;
+
+namespace Umbraco.Web.PublishedCache.NuCache
+{
+    /// <summary>
+    /// Generates a status report for <see cref="PublishedSnapshotService"/>
+    /// </summary>
+    internal class PublishedSnapshotStatus : IPublishedSnapshotStatus
+    {
+        private readonly PublishedSnapshotService _service;
+        private readonly INuCacheContentService _publishedContentService;
+
+        public PublishedSnapshotStatus(PublishedSnapshotService service, INuCacheContentService publishedContentService)
+        {
+            _service = service;
+            _publishedContentService = publishedContentService;
+        }
+
+        /// <inheritdoc/>
+        public virtual string StatusUrl => "views/dashboard/settings/publishedsnapshotcache.html";
+
+        /// <inheritdoc/>
+        public string GetStatus()
+        {
+            _service.EnsureCaches();
+
+            var dbCacheIsOk = _publishedContentService.VerifyContentDbCache()
+                && _publishedContentService.VerifyMediaDbCache()
+                && _publishedContentService.VerifyMemberDbCache();
+
+            ContentStore contentStore = _service.GetContentStore();
+            ContentStore mediaStore = _service.GetMediaStore();
+
+            var cg = contentStore.GenCount;
+            var mg = mediaStore.GenCount;
+            var cs = contentStore.SnapCount;
+            var ms = mediaStore.SnapCount;
+            var ce = contentStore.Count;
+            var me = mediaStore.Count;
+
+            return
+                " Database cache is " + (dbCacheIsOk ? "ok" : "NOT ok (rebuild?)") + "." +
```

Comment:
I would have defined the variables first and returned an interpolated string so this was readable by human beings.

Reply:
Human being are overrated 👽 Yep I agree, this is just migrated code but I can update